filebeat/input/log/input.go

```diff
@@ -78,6 +79,8 @@ type Input struct {
 	meta                map[string]string
 	stopOnce            sync.Once
 	fileStateIdentifier file.StateIdentifier
+	getStatusReporter   input.GetStatusReporter
```

Comment:
Why can't it be just the status reporter itself? why do we need a get function for it?

Reply:
For v1 inputs, the following sequence takes place as per
https://github.com/elastic/beats/blob/1a6318d215517feb4c5c880619db8589d926956b/libbeat/cfgfile/list.go#L131-L175

- `runner` is first created. The `context` is passed **_during_** this step.
    - `SetStatusReporter` is not called yet. 
- If runner is created successfully, `SetStatusReporter` gets called.
- `runner.Run()` gets called and the runner is started
    - `log` input in our case
    
    
 if we remove `GetStatusReporter`,  we need to:
 - Change the field to a public field
 - Instead of passing a function, we need to pass `StatusReporer` by reference 
     - Because it's initialised after runner creation and we don't pass the context while `Run()`. 
 
 I choose the current approach as it is safe to using `Getters\Setters` and dealing with function pointers can be messy.

Reply:
FWIW, for v2 inputs, the following sequence takes place

- runner is first created
- If runner is created successfully, `SetStatusReporter` gets called.
- `runner.Run(ctx)` gets called and the runner is started
     - The context is passed during this step.
     
Hence, we can directly pass the `StatusReporter` value via `context`. Not the case with v1.